File: trailingZeroesInFactorial.py
#!/usr/bin/python
__author__ = 'blithejack'

def find5(x):
    cnt = 0
    while ((x>0) and (x%5 == 0)):
        x = int(x/5)
        cnt += 1
    return cnt

def trailingZeroes(n):
    zeroes = 0
    for i in range(1,n+1):
        zeroes += find5(i)
    return zeroes

if __name__ == "__main__":
    n = 10000
    print("Trailing 0s in "+str(n)+"! is "+str(trailingZeroes(n)))

    assert (2499 == trailingZeroes(10000))

File: trailingZeroesInFactorial.cpp
// blithejack
#include <iostream>
#include <assert.h>

int find5(int x) {
  int cnt = 0;
  while((x > 0) && ((x%5) == 0))
  {
    x = x/5;
    ++cnt;
  }
  return cnt;
}

int trailingZeroes(int n) {
  int zeroes = 0;
  for(int i=1; i<=n; i++)
  {
    zeroes += find5(i);
  }
  return zeroes;
}

int main() {
  int n = 10000;
  std::cout << "Trailing 0s of " << n << "! is " 
    << trailingZeroes(n) << std::endl;
  
  assert(2499 == trailingZeroes(n));
  return 0;
}

